diff --git a/app/build.gradle.kts b/app/build.gradle.kts
index 2e67075a4..895593394 100644
--- a/app/build.gradle.kts
+++ b/app/build.gradle.kts
@@ -101,6 +101,7 @@ android {
 dependencies {
     implementation(libs.androidx.tracing)
     implementation(project(":core:common"))
+    implementation(project(":data:common"))
     implementation(project(":feature:postcapture"))
     // Compose
     val composeBom = platform(libs.compose.bom)
@@ -178,7 +179,11 @@ dependencies {
     implementation(project(":ui:debug"))
 
     // Low Light implementations
-    implementation(project(":core:camera:low-light-playservices"))
+    implementation(project(":data:camera:low-light"))
+    implementation(project(":data:camera:low-light-playservices"))
+
+    // Postprocess implementations
+    implementation(project(":data:camera:postprocess"))
 }
 
 // Allow references to generated code
diff --git a/core/camera/low-light/build.gradle.kts b/core/camera/low-light/build.gradle.kts
index cade65c9e..52776e2fa 100644
--- a/core/camera/low-light/build.gradle.kts
+++ b/core/camera/low-light/build.gradle.kts
@@ -17,8 +17,6 @@
 plugins {
     alias(libs.plugins.android.library)
     alias(libs.plugins.kotlin.android)
-    alias(libs.plugins.kotlin.kapt)
-    alias(libs.plugins.dagger.hilt.android)
 }
 
 android {
@@ -40,12 +38,5 @@ android {
 
 dependencies {
     implementation(libs.camera.core)
-    implementation(libs.dagger.hilt.android)
-    kapt(libs.dagger.hilt.compiler)
     implementation(libs.kotlinx.coroutines.core)
 }
-
-// Allow references to generated code
-kapt {
-    correctErrorTypes = true
-}
diff --git a/core/camera/postprocess/build.gradle.kts b/core/camera/postprocess/build.gradle.kts
index ceedef0b7..38692d709 100644
--- a/core/camera/postprocess/build.gradle.kts
+++ b/core/camera/postprocess/build.gradle.kts
@@ -17,8 +17,6 @@
 plugins {
     alias(libs.plugins.android.library)
     alias(libs.plugins.kotlin.android)
-    alias(libs.plugins.kotlin.kapt)
-    alias(libs.plugins.dagger.hilt.android)
 }
 
 android {
@@ -37,13 +35,3 @@ android {
         jvmToolchain(17)
     }
 }
-
-dependencies {
-    implementation(libs.dagger.hilt.android)
-    kapt(libs.dagger.hilt.compiler)
-}
-
-// Allow references to generated code
-kapt {
-    correctErrorTypes = true
-}
diff --git a/core/common/build.gradle.kts b/core/common/build.gradle.kts
index 185217b27..dce5d45b2 100644
--- a/core/common/build.gradle.kts
+++ b/core/common/build.gradle.kts
@@ -17,8 +17,6 @@
 plugins {
     alias(libs.plugins.android.library)
     alias(libs.plugins.kotlin.android)
-    alias(libs.plugins.kotlin.kapt)
-    alias(libs.plugins.dagger.hilt.android)
 }
 
 android {
@@ -69,6 +67,7 @@ dependencies {
     implementation(libs.androidx.appcompat)
     implementation(libs.kotlinx.atomicfu)
     implementation(libs.androidx.tracing)
+    implementation("javax.inject:javax.inject:1")
 
     testImplementation(libs.junit)
     testImplementation(libs.truth)
@@ -77,11 +76,4 @@ dependencies {
     androidTestImplementation(libs.androidx.junit)
     androidTestImplementation(libs.androidx.espresso.core)
 
-    // Hilt
-    implementation(libs.dagger.hilt.android)
-    kapt(libs.dagger.hilt.compiler)
-}
-
-kapt {
-    correctErrorTypes = true
 }
diff --git a/core/common/src/main/java/com/google/jetpackcamera/core/common/CommonQualifiers.kt b/core/common/src/main/java/com/google/jetpackcamera/core/common/CommonQualifiers.kt
new file mode 100644
index 000000000..20c11cf6d
--- /dev/null
+++ b/core/common/src/main/java/com/google/jetpackcamera/core/common/CommonQualifiers.kt
@@ -0,0 +1,61 @@
+/*
+ * Copyright (C) 2026 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.jetpackcamera.core.common
+
+import javax.inject.Qualifier
+
+/**
+ * Qualifier for the default file path generator.
+ */
+@Target(AnnotationTarget.FUNCTION, AnnotationTarget.VALUE_PARAMETER)
+@Qualifier
+@Retention(AnnotationRetention.BINARY)
+annotation class DefaultFilePathGenerator
+
+/**
+ * Qualifier for the default capture mode override in LocalSettingsRepository.
+ */
+@Qualifier
+@Retention(AnnotationRetention.BINARY)
+annotation class DefaultCaptureModeOverride
+
+/**
+ * Qualifier for the default save mode.
+ */
+@Qualifier
+@Retention(AnnotationRetention.BINARY)
+annotation class DefaultSaveMode
+
+/**
+ * Qualifier for the default coroutine dispatcher.
+ */
+@Qualifier
+@Retention(AnnotationRetention.BINARY)
+annotation class DefaultDispatcher
+
+/**
+ * Qualifier for the IO coroutine dispatcher.
+ */
+@Qualifier
+@Retention(AnnotationRetention.BINARY)
+annotation class IODispatcher
+
+/**
+ * Qualifier for the default application-level coroutine scope.
+ */
+@Qualifier
+@Retention(AnnotationRetention.BINARY)
+annotation class DefaultCoroutineScope
diff --git a/core/settings/build.gradle.kts b/core/settings/build.gradle.kts
index 35c256a56..a5c778046 100644
--- a/core/settings/build.gradle.kts
+++ b/core/settings/build.gradle.kts
@@ -17,8 +17,6 @@
 plugins {
     alias(libs.plugins.android.library)
     alias(libs.plugins.kotlin.android)
-    alias(libs.plugins.kotlin.kapt)
-    alias(libs.plugins.dagger.hilt.android)
 }
 
 android {
@@ -46,14 +44,6 @@ android {
 dependencies {
     implementation(libs.kotlinx.coroutines.core)
 
-    // Hilt
-    implementation(libs.dagger.hilt.android)
-    kapt(libs.dagger.hilt.compiler)
-
     // Domain models
     implementation(project(":core:model"))
 }
-
-kapt {
-    correctErrorTypes = true
-}
diff --git a/core/settings/datastore-prefs/build.gradle.kts b/core/settings/datastore-prefs/build.gradle.kts
index c0ca9f8d4..c4da99bd2 100644
--- a/core/settings/datastore-prefs/build.gradle.kts
+++ b/core/settings/datastore-prefs/build.gradle.kts
@@ -77,6 +77,7 @@ dependencies {
     // Access Model and Settings Interface
     implementation(project(":core:model"))
     implementation(project(":core:common"))
+    implementation(project(":data:common"))
     implementation(project(":core:settings"))
 }
 
diff --git a/data/camera/build.gradle.kts b/data/camera/build.gradle.kts
index 8a541423a..711221739 100644
--- a/data/camera/build.gradle.kts
+++ b/data/camera/build.gradle.kts
@@ -71,6 +71,7 @@ dependencies {
     implementation(project(":core:camera:low-light"))
     implementation(project(":core:camera:postprocess"))
     implementation(project(":core:common"))
+    implementation(project(":data:common"))
     implementation(project(":core:model"))
     implementation(project(":data:settings"))
     implementation(project(":core:settings"))
diff --git a/core/camera/low-light-playservices/build.gradle.kts b/data/camera/low-light-playservices/build.gradle.kts
similarity index 96%
rename from core/camera/low-light-playservices/build.gradle.kts
rename to data/camera/low-light-playservices/build.gradle.kts
index 639fa3e51..d21b0d52a 100644
--- a/core/camera/low-light-playservices/build.gradle.kts
+++ b/data/camera/low-light-playservices/build.gradle.kts
@@ -22,7 +22,7 @@ plugins {
 }
 
 android {
-    namespace = "com.google.jetpackcamera.core.camera.lowlight.playservices"
+    namespace = "com.google.jetpackcamera.data.camera.lowlight.playservices"
     compileSdk = libs.versions.compileSdk.get().toInt()
 
     defaultConfig {
diff --git a/core/camera/low-light-playservices/src/main/AndroidManifest.xml b/data/camera/low-light-playservices/src/main/AndroidManifest.xml
similarity index 100%
rename from core/camera/low-light-playservices/src/main/AndroidManifest.xml
rename to data/camera/low-light-playservices/src/main/AndroidManifest.xml
diff --git a/core/camera/low-light-playservices/src/main/java/com/google/jetpackcamera/core/camera/lowlight/playservices/PlayServicesLowLightBoostAvailabilityChecker.kt b/data/camera/low-light-playservices/src/main/java/com/google/jetpackcamera/core/camera/lowlight/playservices/PlayServicesLowLightBoostAvailabilityChecker.kt
similarity index 100%
rename from core/camera/low-light-playservices/src/main/java/com/google/jetpackcamera/core/camera/lowlight/playservices/PlayServicesLowLightBoostAvailabilityChecker.kt
rename to data/camera/low-light-playservices/src/main/java/com/google/jetpackcamera/core/camera/lowlight/playservices/PlayServicesLowLightBoostAvailabilityChecker.kt
diff --git a/core/camera/low-light-playservices/src/main/java/com/google/jetpackcamera/core/camera/lowlight/playservices/PlayServicesLowLightBoostEffect.kt b/data/camera/low-light-playservices/src/main/java/com/google/jetpackcamera/core/camera/lowlight/playservices/PlayServicesLowLightBoostEffect.kt
similarity index 100%
rename from core/camera/low-light-playservices/src/main/java/com/google/jetpackcamera/core/camera/lowlight/playservices/PlayServicesLowLightBoostEffect.kt
rename to data/camera/low-light-playservices/src/main/java/com/google/jetpackcamera/core/camera/lowlight/playservices/PlayServicesLowLightBoostEffect.kt
diff --git a/core/camera/low-light-playservices/src/main/java/com/google/jetpackcamera/core/camera/lowlight/playservices/PlayServicesLowLightBoostEffectProvider.kt b/data/camera/low-light-playservices/src/main/java/com/google/jetpackcamera/core/camera/lowlight/playservices/PlayServicesLowLightBoostEffectProvider.kt
similarity index 100%
rename from core/camera/low-light-playservices/src/main/java/com/google/jetpackcamera/core/camera/lowlight/playservices/PlayServicesLowLightBoostEffectProvider.kt
rename to data/camera/low-light-playservices/src/main/java/com/google/jetpackcamera/core/camera/lowlight/playservices/PlayServicesLowLightBoostEffectProvider.kt
diff --git a/core/camera/low-light-playservices/src/main/java/com/google/jetpackcamera/core/camera/lowlight/playservices/PlayServicesLowLightBoostModule.kt b/data/camera/low-light-playservices/src/main/java/com/google/jetpackcamera/core/camera/lowlight/playservices/PlayServicesLowLightBoostModule.kt
similarity index 100%
rename from core/camera/low-light-playservices/src/main/java/com/google/jetpackcamera/core/camera/lowlight/playservices/PlayServicesLowLightBoostModule.kt
rename to data/camera/low-light-playservices/src/main/java/com/google/jetpackcamera/core/camera/lowlight/playservices/PlayServicesLowLightBoostModule.kt
diff --git a/core/camera/low-light-playservices/src/main/java/com/google/jetpackcamera/core/camera/lowlight/playservices/PlayServicesLowLightBoostSurfaceProcessor.kt b/data/camera/low-light-playservices/src/main/java/com/google/jetpackcamera/core/camera/lowlight/playservices/PlayServicesLowLightBoostSurfaceProcessor.kt
similarity index 100%
rename from core/camera/low-light-playservices/src/main/java/com/google/jetpackcamera/core/camera/lowlight/playservices/PlayServicesLowLightBoostSurfaceProcessor.kt
rename to data/camera/low-light-playservices/src/main/java/com/google/jetpackcamera/core/camera/lowlight/playservices/PlayServicesLowLightBoostSurfaceProcessor.kt
diff --git a/data/camera/low-light/build.gradle.kts b/data/camera/low-light/build.gradle.kts
new file mode 100644
index 000000000..1af41b83c
--- /dev/null
+++ b/data/camera/low-light/build.gradle.kts
@@ -0,0 +1,52 @@
+/*
+ * Copyright (C) 2026 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+plugins {
+    alias(libs.plugins.android.library)
+    alias(libs.plugins.kotlin.android)
+    alias(libs.plugins.kotlin.kapt)
+    alias(libs.plugins.dagger.hilt.android)
+}
+
+android {
+    namespace = "com.google.jetpackcamera.data.camera.lowlight"
+    compileSdk = libs.versions.compileSdk.get().toInt()
+
+    defaultConfig {
+        minSdk = libs.versions.minSdk.get().toInt()
+    }
+
+    compileOptions {
+        sourceCompatibility = JavaVersion.VERSION_17
+        targetCompatibility = JavaVersion.VERSION_17
+    }
+    kotlin {
+        jvmToolchain(17)
+    }
+}
+
+dependencies {
+    implementation(project(":core:camera:low-light"))
+
+    // Hilt
+    implementation(libs.dagger.hilt.android)
+    kapt(libs.dagger.hilt.compiler)
+}
+
+// Allow references to generated code
+kapt {
+    correctErrorTypes = true
+}
diff --git a/data/camera/low-light/src/main/AndroidManifest.xml b/data/camera/low-light/src/main/AndroidManifest.xml
new file mode 100644
index 000000000..57317bf89
--- /dev/null
+++ b/data/camera/low-light/src/main/AndroidManifest.xml
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+  ~ Copyright (C) 2026 The Android Open Source Project
+  ~
+  ~ Licensed under the Apache License, Version 2.0 (the "License");
+  ~ you may not use this file except in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~      http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+<manifest xmlns:android="http://schemas.android.com/apk/res/android">
+</manifest>
diff --git a/core/camera/low-light/src/main/java/com/google/jetpackcamera/core/camera/lowlight/LowLightBoostModule.kt b/data/camera/low-light/src/main/java/com/google/jetpackcamera/data/camera/lowlight/LowLightBoostModule.kt
similarity index 89%
rename from core/camera/low-light/src/main/java/com/google/jetpackcamera/core/camera/lowlight/LowLightBoostModule.kt
rename to data/camera/low-light/src/main/java/com/google/jetpackcamera/data/camera/lowlight/LowLightBoostModule.kt
index 030d18bb6..cf968bb0c 100644
--- a/core/camera/low-light/src/main/java/com/google/jetpackcamera/core/camera/lowlight/LowLightBoostModule.kt
+++ b/data/camera/low-light/src/main/java/com/google/jetpackcamera/data/camera/lowlight/LowLightBoostModule.kt
@@ -13,8 +13,11 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package com.google.jetpackcamera.core.camera.lowlight
+package com.google.jetpackcamera.data.camera.lowlight
 
+import com.google.jetpackcamera.core.camera.lowlight.LowLightBoostAvailabilityChecker
+import com.google.jetpackcamera.core.camera.lowlight.LowLightBoostEffectProvider
+import com.google.jetpackcamera.core.camera.lowlight.LowLightBoostFeatureKey
 import dagger.Module
 import dagger.Provides
 import dagger.hilt.InstallIn
diff --git a/data/camera/postprocess/build.gradle.kts b/data/camera/postprocess/build.gradle.kts
new file mode 100644
index 000000000..94cde397b
--- /dev/null
+++ b/data/camera/postprocess/build.gradle.kts
@@ -0,0 +1,52 @@
+/*
+ * Copyright (C) 2026 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+plugins {
+    alias(libs.plugins.android.library)
+    alias(libs.plugins.kotlin.android)
+    alias(libs.plugins.kotlin.kapt)
+    alias(libs.plugins.dagger.hilt.android)
+}
+
+android {
+    namespace = "com.google.jetpackcamera.data.camera.postprocess"
+    compileSdk = libs.versions.compileSdk.get().toInt()
+
+    defaultConfig {
+        minSdk = libs.versions.minSdk.get().toInt()
+    }
+
+    compileOptions {
+        sourceCompatibility = JavaVersion.VERSION_17
+        targetCompatibility = JavaVersion.VERSION_17
+    }
+    kotlin {
+        jvmToolchain(17)
+    }
+}
+
+dependencies {
+    implementation(project(":core:camera:postprocess"))
+
+    // Hilt
+    implementation(libs.dagger.hilt.android)
+    kapt(libs.dagger.hilt.compiler)
+}
+
+// Allow references to generated code
+kapt {
+    correctErrorTypes = true
+}
diff --git a/data/camera/postprocess/src/main/AndroidManifest.xml b/data/camera/postprocess/src/main/AndroidManifest.xml
new file mode 100644
index 000000000..57317bf89
--- /dev/null
+++ b/data/camera/postprocess/src/main/AndroidManifest.xml
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+  ~ Copyright (C) 2026 The Android Open Source Project
+  ~
+  ~ Licensed under the Apache License, Version 2.0 (the "License");
+  ~ you may not use this file except in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~      http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+<manifest xmlns:android="http://schemas.android.com/apk/res/android">
+</manifest>
diff --git a/core/camera/postprocess/src/main/java/com/google/jetpackcamera/core/camera/postprocess/PostProcessModule.kt b/data/camera/postprocess/src/main/java/com/google/jetpackcamera/data/camera/postprocess/PostProcessModule.kt
similarity index 88%
rename from core/camera/postprocess/src/main/java/com/google/jetpackcamera/core/camera/postprocess/PostProcessModule.kt
rename to data/camera/postprocess/src/main/java/com/google/jetpackcamera/data/camera/postprocess/PostProcessModule.kt
index d26475400..f7b5f6bdf 100644
--- a/core/camera/postprocess/src/main/java/com/google/jetpackcamera/core/camera/postprocess/PostProcessModule.kt
+++ b/data/camera/postprocess/src/main/java/com/google/jetpackcamera/data/camera/postprocess/PostProcessModule.kt
@@ -13,8 +13,10 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package com.google.jetpackcamera.core.camera.postprocess
+package com.google.jetpackcamera.data.camera.postprocess
 
+import com.google.jetpackcamera.core.camera.postprocess.ImagePostProcessor
+import com.google.jetpackcamera.core.camera.postprocess.ImagePostProcessorFeatureKey
 import dagger.Module
 import dagger.Provides
 import dagger.hilt.InstallIn
diff --git a/data/common/build.gradle.kts b/data/common/build.gradle.kts
new file mode 100644
index 000000000..c95175a5c
--- /dev/null
+++ b/data/common/build.gradle.kts
@@ -0,0 +1,68 @@
+/*
+ * Copyright (C) 2026 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+plugins {
+    alias(libs.plugins.android.library)
+    alias(libs.plugins.kotlin.android)
+    alias(libs.plugins.kotlin.kapt)
+    alias(libs.plugins.dagger.hilt.android)
+}
+
+android {
+    namespace = "com.google.jetpackcamera.data.common"
+    compileSdk = libs.versions.compileSdk.get().toInt()
+
+    defaultConfig {
+        minSdk = libs.versions.minSdk.get().toInt()
+        testOptions.targetSdk = libs.versions.targetSdk.get().toInt()
+        lint.targetSdk = libs.versions.targetSdk.get().toInt()
+
+        testInstrumentationRunner = "androidx.test.runner.AndroidJUnitRunner"
+        consumerProguardFiles("consumer-rules.pro")
+    }
+
+    flavorDimensions += "flavor"
+    productFlavors {
+        create("stable") {
+            dimension = "flavor"
+            isDefault = true
+        }
+    }
+
+    compileOptions {
+        sourceCompatibility = JavaVersion.VERSION_17
+        targetCompatibility = JavaVersion.VERSION_17
+    }
+    kotlin {
+        jvmToolchain(17)
+    }
+}
+
+dependencies {
+    implementation(libs.kotlinx.coroutines.core)
+
+    // Hilt
+    implementation(libs.dagger.hilt.android)
+    kapt(libs.dagger.hilt.compiler)
+
+    // Project dependencies
+    implementation(project(":core:common"))
+}
+
+// Allow references to generated code
+kapt {
+    correctErrorTypes = true
+}
diff --git a/data/common/consumer-rules.pro b/data/common/consumer-rules.pro
new file mode 100644
index 000000000..260809e6c
--- /dev/null
+++ b/data/common/consumer-rules.pro
@@ -0,0 +1 @@
+# Proguard rules for data:common
diff --git a/data/common/src/main/AndroidManifest.xml b/data/common/src/main/AndroidManifest.xml
new file mode 100644
index 000000000..57317bf89
--- /dev/null
+++ b/data/common/src/main/AndroidManifest.xml
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+  ~ Copyright (C) 2026 The Android Open Source Project
+  ~
+  ~ Licensed under the Apache License, Version 2.0 (the "License");
+  ~ you may not use this file except in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~      http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+<manifest xmlns:android="http://schemas.android.com/apk/res/android">
+</manifest>
diff --git a/core/common/src/main/java/com/google/jetpackcamera/core/common/CommonModule.kt b/data/common/src/main/java/com/google/jetpackcamera/data/common/CommonModule.kt
similarity index 63%
rename from core/common/src/main/java/com/google/jetpackcamera/core/common/CommonModule.kt
rename to data/common/src/main/java/com/google/jetpackcamera/data/common/CommonModule.kt
index 998ab951e..da4f7d333 100644
--- a/core/common/src/main/java/com/google/jetpackcamera/core/common/CommonModule.kt
+++ b/data/common/src/main/java/com/google/jetpackcamera/data/common/CommonModule.kt
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2023 The Android Open Source Project
+ * Copyright (C) 2026 The Android Open Source Project
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -13,13 +13,15 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package com.google.jetpackcamera.core.common
+package com.google.jetpackcamera.data.common
 
+import com.google.jetpackcamera.core.common.DefaultCoroutineScope
+import com.google.jetpackcamera.core.common.DefaultDispatcher
+import com.google.jetpackcamera.core.common.IODispatcher
 import dagger.Module
 import dagger.Provides
 import dagger.hilt.InstallIn
 import dagger.hilt.components.SingletonComponent
-import javax.inject.Qualifier
 import javax.inject.Singleton
 import kotlinx.coroutines.CoroutineDispatcher
 import kotlinx.coroutines.CoroutineScope
@@ -46,31 +48,3 @@ class CommonModule {
     fun providesCoroutineScope(@DefaultDispatcher dispatcher: CoroutineDispatcher) =
         CoroutineScope(SupervisorJob() + dispatcher)
 }
-
-@Target(AnnotationTarget.FUNCTION, AnnotationTarget.VALUE_PARAMETER)
-@Qualifier
-@Retention(AnnotationRetention.BINARY)
-annotation class DefaultFilePathGenerator
-
-/**
- * provides the default [CaptureMode] to override in [LocalSettingsRepository]
- */
-@Qualifier
-@Retention(AnnotationRetention.BINARY)
-annotation class DefaultCaptureModeOverride
-
-@Qualifier
-@Retention(AnnotationRetention.BINARY)
-annotation class DefaultSaveMode
-
-@Qualifier
-@Retention(AnnotationRetention.BINARY)
-annotation class DefaultDispatcher
-
-@Qualifier
-@Retention(AnnotationRetention.BINARY)
-annotation class IODispatcher
-
-@Qualifier
-@Retention(AnnotationRetention.BINARY)
-annotation class DefaultCoroutineScope
diff --git a/data/media/build.gradle.kts b/data/media/build.gradle.kts
index 91146f6e8..20f9265f0 100644
--- a/data/media/build.gradle.kts
+++ b/data/media/build.gradle.kts
@@ -89,6 +89,7 @@ dependencies {
 
     // Project dependencies
     implementation(project(":core:common"))
+    implementation(project(":data:common"))
     testImplementation(project(":core:common:testing"))
     androidTestImplementation(project(":core:common:testing"))
 
diff --git a/feature/preview/build.gradle.kts b/feature/preview/build.gradle.kts
index 54ab1c505..8276aa3aa 100644
--- a/feature/preview/build.gradle.kts
+++ b/feature/preview/build.gradle.kts
@@ -138,6 +138,7 @@ dependencies {
     implementation(project(":core:camera"))
     implementation(project(":data:camera"))
     implementation(project(":core:common"))
+    implementation(project(":data:common"))
     implementation(project(":data:media"))
     implementation(project(":data:settings"))
     implementation(project(":core:settings"))
diff --git a/settings.gradle.kts b/settings.gradle.kts
index 103a86882..2cd5e3aaf 100644
--- a/settings.gradle.kts
+++ b/settings.gradle.kts
@@ -39,12 +39,14 @@ include(":core:camera")
 include(":data:camera")
 include(":core:camera:testing")
 include(":core:camera:low-light")
-include(":core:camera:low-light-playservices")
+include(":data:camera:low-light")
+include(":data:camera:low-light-playservices")
 include(":feature:settings")
 include(":data:settings")
 include(":data:settings:testing")
 include(":data:media")
 include(":data:media:testing")
+include(":data:common")
 include(":core:common")
 include(":core:common:testing")
 include(":benchmark")
@@ -62,6 +64,7 @@ include(":core:model")
 include(":ui:uistate:postcapture")
 include(":ui:uistateadapter:postcapture")
 include(":core:camera:postprocess")
+include(":data:camera:postprocess")
 include(":ui:controller")
 include(":ui:controller:impl")
 include(":ui:controller:testing")
diff --git a/ui/components/capture/build.gradle.kts b/ui/components/capture/build.gradle.kts
index d2f9fc5fa..e2f456b45 100644
--- a/ui/components/capture/build.gradle.kts
+++ b/ui/components/capture/build.gradle.kts
@@ -64,6 +64,9 @@ dependencies {
     val composeBom = platform(libs.compose.bom)
     implementation(composeBom)
 
+    // AndroidX Core KTX
+    implementation(libs.androidx.core.ktx)
+
     // Accompanist - Permissions
     implementation(libs.accompanist.permissions)
 
@@ -74,9 +77,6 @@ dependencies {
     implementation(libs.compose.ui.tooling.preview)
     debugImplementation(libs.compose.ui.tooling)
 
-    // Compose - Integration with ViewModels with Navigation and Hilt
-    implementation(libs.hilt.navigation.compose)
-
     // CameraX
     implementation(libs.camera.core)
     implementation(libs.camera.compose)
